nncf/experimental/post_training/algorithms/quantization/min_max/algorithm.py

```diff
@@ -0,0 +1,373 @@
+        """
+        self.weight_quantizer_config = weight_quantizer_config
+        self.activation_quantizer_config = activation_quantizer_config
+        self.number_samples = number_samples
+        self.target_device = target_device
+        self.range_type = range_type
+        self.ignored_scopes = [] if ignored_scopes is None else ignored_scopes
+        self.quantize_outputs = quatize_outputs
+
+    def to_json(self) -> Dict[str, Union[str, float, int]]:
+        """
+        Serialize all MinMaxQuantization parameters to JSON.
+        """
+
+
+class MinMaxQuantization(Algorithm):
+
+    """
+    Post-training MinMaxQuantization algorithm implementation.
+
+    The main purpose of this algorithm to insert FakeQuantizes
+    (or pairs of Quantizer-Dequantizer operations) into the model.
+    This operation is very expressive and allows mapping values from arbitrary input and output ranges.
+    This algorithm projects (discretize) the input values to the low-precision data type
+    using affine transformation (with clamp and rounding) and then re-project discrete values back to the
+    original range and data type.
+    It can be considered as an emulation of the quantization/dequantization process which happens at runtime.
+
+    :param weight_quantizer_config: QuantizerConfig instance for the weights.
+    :param activation_quantizer_config: QuantizerConfig instance for the activations.
+    :param number_samples: The number of the samples for the statistics collection.
+    :param target_device: Target device for the settings of the quantization pipeline.
+    :param range_type: Range types for the quantizer's parameters.
+    :param quantize_outputs: Boolean value that says whether quantize outputs or not.
+    :param ignored_scopes: List of the layers that should not quantized during the process.
+    :param nncf_graph: NNCFGraph class for the algorithm.
+    :param _quantization_target_points: Set of the unique target points.
+    """
+
+    DEFAULT_QCONFIG = QuantizerConfig(num_bits=8,
+                                      mode=QuantizationMode.SYMMETRIC,
+                                      signedness_to_force=None,
+                                      per_channel=False)
+
+    def __init__(self, parameters: MinMaxQuantizationParameters):
+        super().__init__()
+        self.weight_quantizer_config = parameters.weight_quantizer_config \
+            if parameters.weight_quantizer_config is not None else self._get_default_qconfig()
+        self.activation_quantizer_config = parameters.activation_quantizer_config \
+            if parameters.activation_quantizer_config is not None else self._get_default_qconfig()
+        self.number_samples = parameters.number_samples
+        self.target_device = parameters.target_device
+        self.range_type = parameters.range_type
+        self.quantize_outputs = parameters.quantize_outputs
+        self.ignored_scopes = parameters.ignored_scopes
```

Comment:
I think this code is redundant. I suggest to write something like this:

```python
self._params = parameters
```

and use `self._params` like this:

```python
self._params.ignored_scopes
```